app/src/components/notifications/LNotification.vue

```diff
@@ -1,35 +1,77 @@
 <script setup lang="ts">
-import { ref } from "vue";
+import { ref, type FunctionalComponent } from "vue";
 import { CheckCircleIcon, ExclamationCircleIcon } from "@heroicons/vue/24/outline";
 import { XMarkIcon } from "@heroicons/vue/20/solid";
-import type { Notification } from "@/stores/notification";
+import { type Notification, useNotificationStore } from "@/stores/notification";
 
 type Props = {
     notification: Notification;
 };
 
-defineProps<Props>();
+const props = defineProps<Props>();
 
 const show = ref(true);
+
+const icon = ref<FunctionalComponent>();
+
+const { removeNotification } = useNotificationStore();
+
+if (props.notification.icon) {
+    icon.value = props.notification.icon;
+} else {
+    switch (props.notification.state) {
+        case "success":
+            icon.value = CheckCircleIcon;
+            break;
+        case "error":
+            icon.value = ExclamationCircleIcon;
+            break;
+        case "info":
+            icon.value = ExclamationCircleIcon;
+            break;
+        case "warning":
+            icon.value = ExclamationCircleIcon;
+            break;
+    }
+}
+
+const colour = ref<string>("bg-gray-100");
+const textColor = ref<string>("text-gray-400");
```

Comment:
Nitpick: Here we have `colour` (British English) and `Color` (American English). Let's use only one spelling, everywhere else it's American, at least in the case of `color`.

Reply:
Also I think it's confusing to have `colour` which is the background color for the banner, and `textColor` which is only used for the toast. Part of the reason why I propose to split this component into two, see my other comment.